airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py

```diff
@@ -142,7 +142,20 @@ class Events(IncrementalAmplitudeStream):
     state_checkpoint_interval = 1000
     time_interval = {"days": 3}
 
+    # To avoid raising http exceptions by default. (HttpStream class raises http errors by default if set to True)
+    # Export API(Used to get events data) returns 404 when there is no data which is being propogated to UI as error in sync
+    @property
+    def raise_on_http_errors(self) -> bool:
+        """
+        Override if needed. If set to False, allows opting-out of raising HTTP code exception.
+        """
+        return False
+
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        # Read records func calls this func which raises exception when 404 is sent as response
+        if response.status_code == 404:
+            self.logger.warn("No data available for the time range requested.")
```

Comment:
Are we propagating this error to the frontend?

Reply:
No. Only errors are shown in UI (Logic on sources side I guess)